### PR TITLE
allow specifying both to, from & reply_to and fallback to to if reply_to points at nothing

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

**Comment by smn** [comment]
@rudigiesler @KaitCrawford @erikh360 ready for review

File: docs/http_api.rst [hunk 1]
```diff
@@ -252,16 +252,21 @@ Messages
    Send an outbound (mobile terminated) message.
 
    :param str to:
-       the address (e.g. MSISDN) to send the message too. Should be omitted
-       if ``reply_to`` is specified.
+       the address (e.g. MSISDN) to send the message too.
+       If Junebug is configured with ``allow_late_replies`` The ``to``
```
Comment (rudigiesler) [inline docs/http_api.rst]: The config seems to suggest that this should be `allow_expired_replies` instead?

File: junebug/config.py [hunk 2]
```diff
@@ -48,6 +48,12 @@ class JunebugConfig(Config):
         "Maximum time (in seconds) allowed for events to arrive for messages",
         default=60 * 60 * 24 * 2)
 
+    allow_expired_replies = ConfigBool(
+        "If `True` messages with a reply_to that arrive for which the original "
+        "inbound cannot be found (possible of the TTL expiring) are sent as "
+        "normal outbound messages. ",
+        default=False)
```
Comment (rudigiesler) [inline junebug/config.py]: We should also add this to the command line arguments

File: junebug/tests/test_api.py [hunk 3]
```diff
@@ -670,35 +670,81 @@ def test_send_message_additional_properties(self):
...
-        resp = yield self.post('/channels/foo-bar/messages/', {
+
+        properties = self.create_channel_properties(character_limit=100)
+        config = yield self.create_channel_config()
+        redis = yield self.get_redis()
+        channel = Channel(redis, config, properties, id='test-channel')
+        yield channel.save()
+        yield channel.start(self.service)
+
+        self.maxDiff = None
```
Comment (rudigiesler) [inline junebug/tests/test_api.py]: Left over from debugging?

File: junebug/tests/test_api.py [hunk 4]
```diff
@@ -670,35 +670,81 @@ def test_send_message_additional_properties(self):
...
+            allow_expired_replies=True)
+        redis = yield self.get_redis()
+        yield self.stop_server()
+        yield self.start_server(config=config)
+
+        channel = Channel(redis, config, properties, id='test-channel')
+        yield channel.save()
+        yield channel.start(self.service)
+
+        self.maxDiff = None
```
Comment (rudigiesler) [inline junebug/tests/test_api.py]: Same here

File: junebug/tests/test_api.py [hunk 5]
```diff
@@ -670,35 +670,81 @@ def test_send_message_additional_properties(self):
...
-        resp = yield self.post('/channels/foo-bar/messages/', {
-            'from': '+1234',
+        properties = self.create_channel_properties(character_limit=100)
+        config = yield self.create_channel_config()
+        redis = yield self.get_redis()
+        channel = Channel(redis, config, properties, id='test-channel')
+        yield channel.save()
+        yield channel.start(self.service)
+
+        self.maxDiff = None
```
Comment (rudigiesler) [inline junebug/tests/test_api.py]: And here

**Comment by smn** [comment]
@rudigiesler thanks! ready for review again.

File: junebug/tests/test_command_line.py [hunk 6]
```diff
@@ -509,7 +504,7 @@ def test_logging_setup(self):
         is not None, both the stdout logger and a file logger is created'''
         logging_setup(None, None)
         [handler] = logging.getLogger().handlers
-        self.assertEqual(handler.stream.name, '<stdout>')
+        self.assertTrue(handler.stream.name in ['<stdout>', '<fdopen>'])
```
Comment (smn) [inline junebug/tests/test_command_line.py]: This change allows us to use `py.test -x -f` during development. Py tests replaces <stdout> with a fd for capturing test log output

**Pull request merged**